packages/transformers/vue/src/VueTransformer.js

```diff
@@ -0,0 +1,358 @@
+// @flow strict-local
+import {Transformer} from '@parcel/plugin';
+import nullthrows from 'nullthrows';
+import {md5FromObject} from '@parcel/utils';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+import type {Diagnostic} from '@parcel/diagnostic';
+import type {TransformerResult} from '@parcel/types';
+import SourceMap from '@parcel/source-map';
+import semver from 'semver';
+import {basename, extname} from 'path';
+
+const MODULE_BY_NAME_RE = /\.module\./;
+
+// TODO: flow
+export default (new Transformer({
+  canReuseAST({ast}) {
+    return ast.type === 'vue' && semver.satisfies(ast.version, '3.0.0-beta.20');
+  },
+  async parse({asset, options}) {
+    // TODO: This parses the vue component multiple times. Fix?
```

Comment:
Not sure if we can work around this, maybe you could use the `postProcess` function for some of this?

Reply:
All this really does is parse the SFC using Vue's API, so moving it to `postProcess` can't work. I''ll leave this comment as a future optimization.